### ISSUE TITLE
Prometheus exporter produces invalid Prometheus text format

### ISSUE THREAD

**Issue by lchdev** [open-telemetry/opentelemetry-java#5693]
**Describe the bug**

The Prometheus exporter automatically adds `target_info `and `otel_scope_info` metadata metrics. Those metrics are gauges, but in their `#TYPE` line, the type is set to `info`, which is not a valid prometheus type.

As such, the export is not compliant to the spec and will be rejected by some scrappers.

**What did you expect to see?**
```
# TYPE target gauge
# HELP target Target metadata
target_info{*** SOME LABELS***} 1
# TYPE otel_scope_info gauge
# HELP otel_scope_info Scope metadata
otel_scope_info{*** SOME LABELS***} 1
```

**What did you see instead?**
```
# TYPE target info
# HELP target Target metadata
target_info{*** SOME LABELS***} 1
# TYPE otel_scope_info info
# HELP otel_scope_info Scope metadata
otel_scope_info{*** SOME LABELS***} 1
```

**What version and what artifacts are you using?**

Affects latest tag (1.28.0) and main branch.

**Comment by breedx-splk** [comment]
Thanks @lchdev . It helps reviewers quite a bit if you can link to the spec when submitting issues. I think I managed to track it down, though. [Is this the relevant section of the prom text spec](https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#comments-help-text-and-type-information)? It looks like it to me.

**Comment by dashpole** [comment]
info is a valid type in OpenMetrics, but not in Prometheus.  https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#info

**Comment by breedx-splk** [comment]
I think this PR intended to point at this core repo and not the fork itself: https://github.com/lchdev/opentelemetry-java/pull/1

**Comment by lchdev** [comment]
> info is a valid type in OpenMetrics, but not in Prometheus. https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#info

Indeed, and as such, it should be converted when exposed by the Prometheus endpoint. 

It's common practice to have "info" metrics exposed as gauged with a fixed value in Prometheus, so I proposed this simple change for those 2 hard-coded metrics in the Prometheus Serializer.

The same issue could however arise when trying to export metrics with types not defined in the Prometheus spec. I actually did not check the behavior in this case (I only have basic types such as gauges and counters to deal with). Choices would be to either convert them (if possible) or drop them ?

It's also worth noticing that the Prometheus server itself won't care about metric types, only some tooling / scrappers will. 

> I think this PR intended to point at this core repo and not the fork itself: https://github.com/lchdev/opentelemetry-java/pull/1

Yes indeed, I should have paid attention :/ 

I closed the [original PR](https://github.com/lchdev/opentelemetry-java/pull/1) for now and will recreate it to upstream.
